### DISCUSSION TITLE
Subject token limit

### DISCUSSION THREAD

**Discussion by albionb96** [nats-io/nats-server#5097] [Q&A] [answered]
In the documentation it states that: "It is recommended to keep the maximum number of tokens in your subjects to a reasonable value of 16 tokens max". And I wanted to ask what is actually meant with that:
1) Is it meant that the number of characters between the two full-stops shouldn't exceed 16  time.<16 characters>.east or,
2) is it meant that the number of the domains itself shouldn't exceed 16: time.us.east.atlanta.5.6.7.8.9.10.11.12.13.14.15.16

I believe the second option is the case, but I just wanted to make sure.

**Comment by MauriceVanVeen** [comment, marked as answer]
Indeed number 2, also as described here: https://docs.nats.io/nats-concepts/subjects#characters-allowed-for-subject-names

> Special characters: The period . (which is used to separate the tokens in the subject) and * and also > (the * and > are used as wildcards).

The `.` separates tokens. Should indeed keep that at a reasonable value and not go overboard with the amount of tokens. That way the server can guarantee performance of subjects up to that maximum, although it's probably not strictly enforced it would have performance impact.
